test/rust-test-util/TestResultReportingClient/src/models/test_result_dto.rs

```diff
@@ -0,0 +1,98 @@
+use chrono::{DateTime, Utc};
+use serde::{Deserialize, Serialize};
+
+use crate::MessageTestResult;
+
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub struct TestOperationResultDto {
+    #[serde(rename = "source")]
+    source: String,
+    #[serde(rename = "result")]
+    result: MessageTestResult,
+    #[serde(rename = "type")]
+    _type: u8,
+    #[serde(rename = "createdAt")]
+    created_at: String,
+}
+
+impl TestOperationResultDto {
+    pub fn new(
+        source: String,
+        result: MessageTestResult,
```

Comment:
We technically should support more than just MessageTestResult, but that is all I see us needing for the forseeable future.